business/dashboards.go

```diff
@@ -0,0 +1,148 @@
+package business
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+	"github.com/kiali/kiali/prometheus"
+)
+
+// DashboardsService deals with fetching dashboards from k8s client
+type DashboardsService struct {
+	prom prometheus.ClientInterface
+	mon  kubernetes.KialiMonitoringInterface
+}
+
+// NewDashboardsService initializes this business service
+func NewDashboardsService(mon kubernetes.KialiMonitoringInterface, prom prometheus.ClientInterface) DashboardsService {
+	return DashboardsService{prom: prom, mon: mon}
+}
+
+// GetDashboard returns a dashboard filled-in with target data
+func (in *DashboardsService) GetDashboard(params prometheus.CustomMetricsQuery, template string) (*models.MonitoringDashboard, error) {
+	dashboard, err := in.mon.GetDashboard(params.Namespace, template)
```

Comment:
"in.mon.GetDashboard" fetches the "definition" of the dashboard, meanwhile "DashboardsService.GetDashboard" fetches the dashboard populated with data. Am I right ?
If so I feel more easy to read if the "in.mon.GetDashboard" -> "in.mon.GetDashboardDef/Spec/Whatever" just to differentiate.

Reply:
Well, that's just layers doing stuff at their levels... we have the same for GetService, GetNamespace, probably several others... should we change everything?